Documentation/operation docs rework (#3074)
scality/metalk8s: docs/operation/cluster_monitoring/prometheus.rst

Prometheus
==========

In a MetalK8s cluster, the Prometheus service records real-time metrics in a

> time series database. Prometheus can query a list of datasources called

Reviewer: ```suggestion
time series database. Prometheus can query a list of data sources called
```

> `exporters` at specific polling frequency, and aggregate this data across the

Reviewer 2: double backtick

Reviewer: I consider this a reference to a term, not code; hence my suggested quotation marks.

> `exporters` at specific polling frequency, and aggregate this data across the

Reviewer: ```suggestion
"exporters" at a specific polling frequency, and aggregate this data across the
```

various sources.

> Prometheus uses a special language *Prometheus Query Language* (PromQL),

Reviewer: Insert a line if you want a paragraph break. 

```suggestion
Prometheus uses a special language, *Prometheus Query Language* (PromQL),
```

to write alerting and recording rules.

Default Alert Rules
-------------------

Alert rules enable a user to specify a condition that must occur before an

> external system like slack is notified. For example, MetalK8s administrators

Reviewer: ```suggestion
external system like Slack is notified. For example, a MetalK8s administrator
```

> who want to raise an alert for any node that is unreachable for a duration

Reviewer: ```suggestion
might want to raise an alert for any node that is unreachable for
```

> >1 minutes.

Reviewer: more than one minute.

or 

\> 1 minute.

> Out-of-the-box, MetalK8s ships with preconfigured alert rules. These are

Reviewer: ```suggestion
MetalK8s ships with preconfigured alert rules, which are
```
If you think "Out of the box" merits inclusion, remove the hyphens.

written as PromQL queries.
The table below outlines all the preconfigured alert rules exposed from
a newly deployed MetalK8s cluster.

> For predefined alert rules customization, refer to

Reviewer: ```suggestion
To customize predefined alert rules, see
```

:ref:`csc-prometheus-customization`.

> .. csv-table:: Default Prometheus Alerting rules

Reviewer: ```suggestion
.. csv-table:: Default Prometheus Alerting Rules
```

WARNING: If you proceed with CSV tables, you do so at hazard of breaking PDF output. Please test this (I could be wrong) to be sure it outputs to PDF, and understand that you may be committing us to abandoning PDF for this product. 

Frankly, I would love to be rid of PDF, but this is not my decision!

Author: I'm not the one who came up with the CSV tables. It was someone else's doing, and I won't be changing those. At least not in this PR, as it is not the aim of it. The main focus is the reorganization of the guide 😉

Reviewer 3: Good point, will open a ticket to verify PDF output with this piece.
Also, I think the full specification of deployed rules would be a good candidate to only include in a "Reference" guide, so we can keep "Operation" more compact. Another topic for another PR, I guess :)

Reviewer 3: Opened #3188
